src/praisonai/praisonai/llm/registry.py

```
@@ -268,10 +268,43 @@ def _register_builtin_providers(registry: LLMProviderRegistry) -> None:
+            self.provider_id = "litellm"
+            self.model_id = model_id
+            self.config = config or {}
+
+        def generate(self, prompt: str, **kwargs):
+            try:
+                import litellm  # lazy
+            except ImportError:
+                raise ImportError(
+                    "LiteLLM is required for built-in providers. "
+                    "Install with: pip install litellm"
+                )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)